hack/Dockerfile.buildproto

```diff
@@ -0,0 +1,36 @@
+#   Copyright The containerd Authors.
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#       http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+ARG GO_VERSION=1.24
+
+FROM golang:${GO_VERSION}-bullseye AS builder
```

Comment:
This is also needed for the existing Dockerfile; we probably should have a separate build-arg for the debian variant (bullseye may no longer be provided now that trixie is out). Can be something similar to https://github.com/moby/moby/blob/b8897d91db618217dc10a6bf808ed3cf60c35839/Dockerfile#L3-L5

```
ARG GO_VERSION=1.24.6
ARG BASE_DEBIAN_DISTRO="bookworm"
ARG GOLANG_IMAGE="golang:${GO_VERSION}-${BASE_DEBIAN_DISTRO}"
```